### ISSUE TITLE
Facets and record view changes for event data

### ISSUE THREAD

**Issue by javier-molina** [AtlasOfLivingAustralia/extended-data-model#29]
This issue takes an opinionated approach to move some fields around in the Biocache Hub to make more intuitive for users to find event information.

### Facets
1. Add Event section to facets just below Occurrence

![Screen Shot 2022-03-22 at 9 32 42 am](https://user-images.githubusercontent.com/8326394/159377178-cab85958-9495-408e-8285-e731a5993c30.png)

For the filters configuration:

2. Create new event section.

3. Move Month, Year, Date Precision, Year (by decade) and Event ID from Occurrence Section to new Event section
4.  Move Dataset name from Attribution to new Event section
5.  Rename " Dataset name" to  "Dataset /Survey name"
6. Add Parent Event ID and Field number fields to facets
7. Proposed order of fields:

|Left|Right|
|-----|-----|
|Dataset /Survey name|Month|
|Parent Event ID|Year|
|Field number|Date Precision|
|Event ID|Year (by decade)|

   
![Screen Shot 2022-03-22 at 9 56 22 am](https://user-images.githubusercontent.com/8326394/159377192-03b52e45-1dc0-4393-bd30-bcb1c17baece.png)

### Record view page

1. Move Field number and Dataset  / Survey name from Dataset section to Event section
2. Add Event ID and Parent Event ID
3. The proposed order of fields in the Event section is:

- Dataset  / Survey name
- Parent Event ID
- Field number
- Event ID
- Existing list of event fields

![Screen Shot 2022-03-22 at 9 37 24 am](https://user-images.githubusercontent.com/8326394/159377212-ee2d5267-ffb5-4f9e-bf6f-f278f1dc686e.png)

![Screen Shot 2022-03-22 at 9 36 14 am](https://user-images.githubusercontent.com/8326394/159378073-b0eee413-f5b0-4e38-9193-ee0d82b2262d.png)

**Comment by elywallis** [comment]
Observation 1: Facet groups and the order the data is presented in record view do not match. Whatever we do at the moment is window-dressing because users will still have the challenge that the data is presented in a different order. By that I don't mean 'needs a few tweaks' - there are major differences. This means that users are always having to hunt for where a particular data element is on the page because there is no consistency between the search results page and the record view page. If we were to do this properly we would do a major reorganisation to both the facet list order and the record view page.

Observation 2: Field number is already repeated unncecessarily, so having it appear only once under Event is fine. A number of other fields are also repeated (such as date identified, sampling protocol). We would need to sample a number of different records across data resources to find all the duplication. (You'd do this as part of fixing observation 1)

Observation 3. Names of collectors are still processed poorly. The screenshot above is okay but see https://biocache.ala.org.au/occurrences/e26ba0d8-8945-4f2e-b336-2553f2ea975e for example. We also need to prepare for the time when collections supply an ORCID, VIAF or WikiData ID instead of a name. 

Observation 4. There is a lack of agreement between museums and herbaria about which Event fields to use. Museums seem to use datasetName and eventID or fieldNumber. Herbaria use parentEventID instead of datasetName. Thus the better order for the Event grouping might be:
Dataset / Survey name
Parent Event ID
Field number
Event ID
...Existing list of event fields

**Comment by javier-molina** [comment]
@adam-collins  @nielsklazenga, @nickdos, @peggynewman
Could I have your comments re: this change before we implement any changes?

The comment above from Ely is more general but Observation 4 is the one that applies to this work.

**Comment by nickdos** [comment]
Changes look good to me 👍
We might need to get feedback from AVH folks (@elywallis  can advise "who") on whether AVH should also incorporate these changes.

**Comment by elywallis** [comment]
I'd take advice from @nielsklazenga in the first instance and Anne Fuchs can put it to HISCOM in the second

**Comment by nielsklazenga** [comment]
I agree with the changes @javier-molina suggests, except the confounding of `datasetName` and survey name. A survey is an Event, not a Dataset. Just because there is no Darwin Core term for it (yet) does not mean we can just co-opt a completely unrelated one. I personally also think `recordedBy` belongs to the Event, but Darwin Core has it in Occurrence, so better keep it there.

I also agree with @elywallis's observations. There is more needed than just moving a few filters around, but this is a start and creating an Event filter group brings the search result page closer to the record detail page, which already has the Event.

**Comment by javier-molina** [comment]
@nielsklazenga 

Not sure we can do much about the co-opting of Dataset / Survey at this stage. If I understand correctly it has been there in Biocache for a while and a more sensible fix would require changes to the index. For this, we are keeping changes at the UI level while we focus most of our effort in the new event processing system. Will this be a dealbreaker or do you see a way around not touching the index?

![Screen Shot 2022-04-07 at 2 57 07 pm](https://user-images.githubusercontent.com/8326394/162123123-55c71b49-80eb-4c7b-a267-91f7cb16fe1f.png)

**Comment by nielsklazenga** [comment]
Looks to me it is just a matter of changing the description of the term `datasetName`. We cannot stop people from delivering survey names as `datasetName` and there is actually nothing wrong with that, as the scope of a dataset can be a single survey. We should, however, not force people to deliver survey name as `datasetName`.

I think the easiest (and best) way to deal with this is leave the `datasetName` property completely out of it and have a clean new property for 'survey name'. This requires adding a term to the index of course, which is a change, but that should not be an issue I think. We can then propose to have `eventName` and `parentEventName` to Darwin Core.

@elywallis can confirm that we have had this discussion around expeditions in the collections community for more then ten years, and around BushBlitz for about eight years. Surveys are no different to expeditions (in the sense that they are both Events). There is just no way to deal with it properly in Darwin Core. I think the Extended Data project is a good opportunity to try to fix this once and for all.

**Comment by nielsklazenga** [comment]
@javier-molina 
Looks to me it is just a matter of changing the description of the term `datasetName`. We cannot stop people from delivering survey names as `datasetName` and there is actually nothing wrong with that, as the scope of a dataset can be a single survey. We should, however, not force people to deliver survey name as `datasetName` by telling people it is the right way to deliver survey names, which is what you do by moving `datasetName` into the Event group. You will also always keep the problem that `datasetName` is not necessarily a survey name and `datasetName` is extremely messy in ALA and I think that will show in the facets, so it would have to be cleaned up before it can be used anyway. 

I think the easiest (and best) solution is leave the `datasetName` property alone and have a clean new property for 'survey name'. This requires adding a term to the index of course, which is a change, but that should not be an issue I think (and is  definitely less of an issue than cleaning up `datasetName`). We can then propose to have `eventName` (and maybe `parentEventName`) added to Darwin Core.

@elywallis can confirm that we have had this discussion around expeditions in the collections community for more then ten years, and around BushBlitz for about eight years. Surveys are no different to expeditions (in the sense that they are both Events). There is just no way to deal with it properly in Darwin Core. I think the Extended Data project is a good opportunity to try to fix this once and for all.

**Comment by elywallis** [comment]
I can indeed confirm that we have had a discussion around expeditions for more than 10 years. I can also confirm that despite this long duration we have not come up with an agreed approach for documentation. 

I've also just had a look at the entries in the current DatasetName field. I can't see all of them but from what I can it does look like there is a mix of entities you'd describe as "datasets" (e.g. iNat research grade observations) and things you'd describe as surveys or expeditions (such as Museums Victoria's Port Phillip Bay Survey records). If we do set up a clean property for 'survey name' that's going to be problematic for sharing data through to GBIF and I don't want the collections who use DatasetName for surveys to lose data. Perhaps it's a matter of going to the museums collections and asking them to remap their data to ParentEventID to standardise with the herbaria?

**Comment by nielsklazenga** [comment]
The problem is that with `datasetName` you just do not know whether it is a survey name or not. It is much easier to merge two fields together than to split one field in two. So, if I would get a dataset with a column which I know has survey names, even if the column is called 'datasetName`, I would want that to go into a dedicated 'survey name' field rather than `datasetName`. As long as we have the data separately, we can still deliver the data to GBIF as `datasetName`, or any other term they want it in.

Also, if in museum datasets `datasetName` is always survey names, museums do not need to change that, as the pre-ingestion framework will be able to handle that (pretty sure). I myself use an IPT, so I am not able to deliver a non-Darwin Core, so I deliver that data as `eventRemarks` (and `parentEventID`, which is what HISCOM agreed on with Miles and Michael at the time). Of course museums – and definitely surveys – might have more uses for `eventRemarks`, so this is not an option for them, if they want to do something with 'survey name'.

It would be good to get @peggynewman's perspective on this, as I do not really want to make promises about what the Data team can and cannot handle.

**Comment by nielsklazenga** [comment]
Overall, by the way, @javier-molina, I agree with all the changes you suggest, so we would want them implemented in AVH as well. We do not have the `datasetName` filter in AVH, so whatever you do will not be a deal breaker for us. We would very much like a filter for surveys, which in our world are field trips or expeditions (and we do not use `datasetName`), but you might consider that another issue for another time.

**Comment by nielsklazenga** [comment]
Also noting that with the pipelines infrastructure, as you know, we can very easily reload datasets, possibly with changed mappings, without needing the data redelivered, as we keep the provided Darwin Core Archives (or the Darwin Core Archives we create from the provided data). New DwCAs for GBIF are also created and harvested every month. So we can deal with the expeditions and field trips separately later.

I have the feeling though that for the Biocache as a whole, the filter on `datasetName` in the Event group will be problematic and there is no time like the present. We can just try it out first though.

**Comment by elywallis** [comment]
Quick scan through the data - summarising records that have an entry under DatasetName. There are 4560 unique entries. 2230 of these entries contain the word "survey", a further 192 contain the word "expedition" and 33 contain the word "bushblitz" or "bush blitz". A quick visual scan would suggest that there are many more names of field trips or surveys in there as well, just not using one of those keywords. Conclusion is that if we go with @nielsklazenga suggestion to pull survey names out to a new field then that will actually largely empty out the datasetName field of values. 
@nielsklazenga if the herbaria are recording the name of a BushBlitz would you put it in ParentEventID? I'd quite like to try to get all these survey names in one element even if it means asking the museums to change to using ParentEventID and even if that's still not a great place to put the data

**Comment by nielsklazenga** [comment]
@elywallis , yes, we put the canonical strings in `parentEventID`, which is where ALA can get it from. This is not an ideal solution either; ideally we would have Events with IDs we could reference, even if it would be a single webpage with bookmarks. That's why I also deliver it as `eventRemarks` (with some extra data like the start and end dates), which I think is the best spot for it in Darwin Core, but that is a remarks field of course, so you cannot do anything with it except display it.

My proposal, by the way, maybe not when I was writing it, but in my mind on my ride home, is that this 'survey name' term is an extra term, which will be populated from either `datasetName` or `parentEventID`, but the data will stay in those terms as well, so we do not have to change anything to the extraction of DwCAs for GBIF, but, if you want to use it as an opportunity to clean up the `datasetName` term that is fine with me too. I personally think `datasetName` is not super useful in the ALA context, as I use it interchangeably with `data resource`, and the data needs to go somewhere as long as there is no dedicated term in Darwin Core.

I suspect that, if we link the values of `datasetName` back to the data resources, we'll find that within a data resource it will have been used consistently for surveys (or expeditions) or something else. We already know that museum collections have used it consistently for expeditions/field trips and herbarium collections have not used it at all, so that can be the start. And the data resources that are known to comprise survey data, of course, if they do not already use a separate term for it. The TERN standard might have a term; ABCD does (two even).

I think it would be good, towards the end of the project, to write a proposal to add a term to Darwin Core, since we've got such a good use case.

**Comment by nielsklazenga** [comment]
Sorry, missed the most important part. Yes, if it is easy for museums to put the data in `parentEventID` that would be good, as we know that is not used for anything else within ALA. We can then also work on creating the Events (at some point in the future), so that they can be proper eventIDs.

**Comment by peggynewman** [comment]
We're dealing with a legacy issue with datasetName - that is, many folk have used it to put the event name in. I think they key field that we want to be working with and encouraging data providers to use in future is **eventID**. THat's the field that GBIF have indexed and use to group occurrences eg see this event page on GBIF for a MV survey: https://www.gbif.org/dataset/39905320-6c8a-11de-8226-b8a03c50a862/event/cpbs-n%2032

**Comment by peggynewman** [comment]
I agree with Niels about datasetName. Maybe surveyName is a good term for DwC, but I would prefer to call it something like 'eventName', because I would hope that it would have a 1:1 relationship with eventID and would prefer that was obvious. Where is @charvolant on this field naming - isn't this part of the model development? I would hope the model was being developed from the back end rather than the front.

**Comment by peggynewman** [comment]
Also the [samplingEffort](http://rs.tdwg.org/dwc/terms/samplingEffort) and [samplingProtocol](http://rs.tdwg.org/dwc/terms/samplingProtocol) DwC terms are tightly bound to Events.

**Comment by elywallis** [comment]
@peggynewman the data that museums put into datasetName is closer to parentEventID (except not an ID) than to eventID. In the example above GBIF has correctly mapped CPBS-N 32 to EventID. The original record from MV will have CPBS-N 32 in FieldNumber. The information in datasetName is CPBS-N. CPBS-N was a marine expedition to the Great Australian Bight, where they sampled at a number of sites (numbered up from 1 and including site 32). There is also CPBS-S, CPBS-A, -B, -C, -0 all representing separate sampling trips. 
You'll find similar sets of 'overall expedition or field trip name or acronym' in datasetName, individual sampling locations in FieldNumber right through MV's data and presumably other museums (I haven't checked). If you want to check other examples, have a look for SLOPE (survey off the Australian continental shelf) and PPBES (surveys done in Port Phillip Bay).

**Comment by elywallis** [comment]
and I agree that samplingEffort and samplingProtocol are tightly bound to Events

**Comment by elywallis** [comment]
But as for 1:1 relationship between surveyName and eventID. No, there should be a 1:1 relationship between **surveyName** and **parentEventID** if you're supplying surveyName as a text string (Lake Condah BushBlitz) and parentEventID as a resolvable ID. 
There's a one:many relationship between parentEventID and eventID. That is, lots of individual sampling events or sites or stations (in marine parlance) within the one field trip or expedition or voyage.

**Comment by nielsklazenga** [comment]
Okay, I just deleted what I was writing, as @elywallis beat me to it. Most surveys will also be parent events (they will often not have names though), as they can span multiple times and locations, all of which will be individual events. We are talking about a recursive or inclusion relationship here, so the 1:1 relationship between survey and either `parentEventID` or `eventID` does not really apply, as Events can be more deeply nested. It is more like the relationship between `country` and `stateProvince` or `county` or `municipality`. I had proposed `eventName` for the Darwin Core term, by the way, but as long as there is no Darwin Core term, I think it might be better to name it something else to avoid confusion. However, all we are saying at this point is that we should not just co-opt `datasetName`, but need a new term.

To be fair, @javier-molina wants to avoid changing the index, aka changing the model. We are just running into a situation where we think the model is not sufficient, aka encountered a term that cannot be mapped to Darwin Core. But we are looking at everything, not just surveys. This might have come up in this particular issue, which was really about something else, but it is something that @elywallis and I and the collections community have been hitting our heads against for over ten years. This is just a good opportunity to think about a solution. I think models should be built from the domain, by the way, not from what the data looks like. I think what the Extended Data project is doing is broadening the domain, so the model might have to be altered as well. Models are also very much a front-end thing. The back-end might look completely different. But I agree, it is much easier if they look more or less the same.

**Comment by javier-molina** [comment]
I had a chat with @nielsklazenga, to try to summarise the trail of messages in the discussion for this issue:

For the purposes of this issue, we are keeping changes to Biocache to the bare minimum and the description section already contains the feedback.

The changes to the data model like the missing DwC term for event name are something that we need to track in the stream of work to produce a new system for event data processing, @charvolant has been driving the design of the model and is aware of this discussion.

**Comment by qifeng-bai** [comment]
@javier-molina I think we should check if Event exits, before we move dataset / field information to  Event block

**Comment by qifeng-bai** [comment]
PR on Biocache-hubs:
https://github.com/AtlasOfLivingAustralia/biocache-hubs/pull/549

PR on ALA-install: An external Facets config:
https://github.com/AtlasOfLivingAustralia/ala-install/commit/85773cc850f70407991c38ef578ba2df3004e0f6

**Comment by qifeng-bai** [comment]
@elywallis and @nielsklazenga  It has been deployed to biocache-test.ala.org.au, but it has only limited datasets available

**Comment by javier-molina** [comment]
Fixed!

> @qifeng-bai my observations for the filter configuration:
> 
> ![Screen Shot 2022-06-03 at 2 15 20 pm](https://user-images.githubusercontent.com/8326394/171785403-bd8e4138-a57e-4164-ae8c-08490660e0d1.png)
> 
> Could you remove duplicate entry "Year (by decade)" in Occurrence section?
> 
> I can see Dataset name is also duplicated but I'd like to understand first, if possible, how it appeared there, I can see it in prod but it was not in my original screenshot in this issue description.
> 
> Finally "Dataset name" in the Event section still needs to be renamed to "Dataset/Survey name".
> 
>

**Comment by qifeng-bai** [comment]
@javier-molina  Fixed. Please test again

**Comment by javier-molina** [comment]
@qifeng-bai 

See
https://biocache-test.ala.org.au/occurrences/28eaa9e4-b34d-4657-9ec4-e7934fcf07bc

![Screen Shot 2022-06-06 at 10 21 24 am](https://user-images.githubusercontent.com/8326394/172076662-d99d5ef8-39b3-439f-937c-c2a9b3951d5d.png)

No eventID or parentEventId are displayed.

And
https://biocache-test.ala.org.au/occurrences/70ab6069-7d46-4bc6-b854-0d840f8d8741

![Screen Shot 2022-06-06 at 10 28 52 am](https://user-images.githubusercontent.com/8326394/172076977-8fcd5bdf-3c33-4353-8b84-23ff0a9e9c7c.png)

Dataset / Survey name is still appearing in the Daset section not event one.

**Comment by qifeng-bai** [comment]
@javier-molina Seems  biocache-ws-test does not return eventID / parent event ID etc.  I am checking on it

https://biocache-ws-test.ala.org.au/ws/occurrence/62b8b68d-1920-43d3-a894-3b1e7110759d

**Comment by nickdos** [comment]
biocche-ws-test needs to have a new SOLR index created for it - I don't think that has been done. As mentioned a few times, the test SOLR index appears to be corrupted in some way.

**Comment by qifeng-bai** [comment]
@nickdos Thanks.  Seems Biocache prod does not return event id / dataset name as well.  I have added them in.

**Comment by javier-molina** [comment]
Works for me.

@elywallis any further comments?

@qifeng-bai feel free to move it to done once the peer review has been performed, if not already done.
